actions/testabilities/fixture_spvwallet_application.go

```diff
@@ -0,0 +1,179 @@
+	ForAdmin() *resty.Client
+	// ForUser returns a new http client that is configured with the authentication with the xpub of the sender user.
+	ForUser() *resty.Client
+}
+
+type appFixture struct {
+	config             *config.AppConfig
+	engine             engine.ClientInterface
+	t                  testing.TB
+	logger             zerolog.Logger
+	transport          testServer
+	dbConnectionString string
+	dbConnection       *sql.DB
+}
+
+func Given(t testing.TB) SPVWalletApplicationFixture {
+	f := &appFixture{
+		t:      t,
+		logger: tester.Logger(t),
+		config: getConfigForTests(),
+	}
+
+	f.initDbConnection()
+
+	return f
+}
+
+func (f *appFixture) NewTest(t testing.TB) SPVWalletApplicationFixture {
+	return newOf(*f, t)
+}
+
+func newOf(f appFixture, t testing.TB) *appFixture {
+	f.t = t
+	f.logger = tester.Logger(t)
+	return &f
+}
+
+func (f *appFixture) StartedSPVWallet() (cleanup func()) {
+	return f.StartedSPVWalletWithConfiguration()
+}
+
+func (f *appFixture) StartedSPVWalletWithConfiguration(opts ...ConfigOpts) (cleanup func()) {
+	for _, opt := range opts {
+		opt(f.config)
+	}
+
+	options, err := f.config.ToEngineOptions(f.logger)
+	require.NoError(f.t, err)
+
+	f.engine, err = engine.NewClient(context.Background(), options...)
+	require.NoError(f.t, err)
+
+	f.registerUsersFromFixture()
+
+	s := server.NewServer(f.config, f.engine, f.logger)
+	f.transport.handlers = s.Handlers()
+
+	return func() {
+		err := f.engine.Close(context.Background())
+		require.NoError(f.t, err)
+	}
+}
+
+func (f *appFixture) HttpClient() SPVWalletHttpClientFixture {
+	return f
+}
+
+func (f *appFixture) ForAnonymous() *resty.Client {
+	c := resty.New()
+	c.OnError(func(_ *resty.Request, err error) {
+		f.t.Fatalf("HTTP request end up with unexpected error: %v", err)
+	})
+	c.GetClient().Transport = f.transport
+	return c
+}
+
+func (f *appFixture) ForAdmin() *resty.Client {
+	c := f.ForAnonymous()
+	c.SetHeader("x-auth-xpub", config.DefaultAdminXpub)
+	return c
+}
+
+func (f *appFixture) ForUser() *resty.Client {
+	c := f.ForAnonymous()
+	c.SetHeader("x-auth-xpub", fixtures.Sender.XPub())
+	return c
+}
+
+func (f *appFixture) registerUsersFromFixture() {
+	opts := f.engine.DefaultModelOptions(engine.WithMetadata("source", "fixture"))
+
+	for _, user := range fixtures.InternalUsers() {
+		_, err := f.engine.NewXpub(context.Background(), user.XPub(), opts...)
+		require.NoError(f.t, err)
+
+		for _, paymail := range user.Paymails {
+			_, err := f.engine.NewPaymailAddress(context.Background(), user.XPub(), paymail, paymail, "", opts...)
+			require.NoError(f.t, err)
+		}
+	}
+}
+
+// initDbConnection creates a new connection that will be used as connection for engine
+func (f *appFixture) initDbConnection() {
+	hex, err := utils.RandomHex(8)
+	require.NoErrorf(f.t, err, "cannot generate random hex for sqlite inmemory db name")
+
+	f.dbConnectionString = "file:" + hex + "?mode=memory&cache=shared"
```

Comment:
Consider not using `cache=shared` to preserve each test cases separation.

Reply:
maybe you're right, maybe there is no good reason to use a cache=shared